common/webhook.cc

```diff
@@ -46,81 +58,103 @@ void WebHookRunner::setMaxConns(unsigned int max_conns)
 // synchronously run the ping command for the hook
 bool WebHookRunner::pingHook(std::shared_ptr<const WebHook> hook, std::string error_msg)
 {
-  auto cc = getConnection(hook); // this will only return once it has a connection
-
-  if (auto ccs = cc.lock())
-    return _runHook("ping", hook, std::string(), ccs);
-  else
-    return(false);
+  MiniCurlMulti mcm;
+  
+  _addHook("ping", hook, std::string(), mcm);
+  std::vector<WebHookQueueItem> wqi = { {std::string("ping"), hook, std::make_shared<std::string>(std::string())} };
+  return _runHooks(wqi, mcm);
 }
 
 // asynchronously run the hook with the supplied data (must be a string in json format)
 void WebHookRunner::runHook(const std::string& event_name, std::shared_ptr<const WebHook> hook, const std::string& hook_data)
 {
   std::string err_msg;
 
+  vdebuglog("runHook: event %s hook id %d", event_name, hook->getID());
   if (!hook->validateConfig(err_msg)) {
     errlog("runHook: Error validating configuration of webhook id=%d for event (%s) [%s]", hook->getID(), event_name, err_msg);
   }
   else {
-    auto cc = getConnection(hook);
-    if (auto ccs = cc.lock())
-      p.push(_runHookThread, event_name, hook, hook_data, ccs);
+    WebHookQueueItem wqi = {
+      event_name,
+      hook,
+      std::make_shared<std::string>(hook_data)
+    };
+    {
+      std::lock_guard<std::mutex> lock(queue_mutex);
+      if (queue.size() > max_queue_size) {
```

Comment:
this seems to allow max_queue_size+1 item in queue

Reply:
true